### PR TITLE
Add `lookup_self_v6`

### PR DESCRIPTION
**Pull request by talhahwahla**

### CODE REVIEW

**Comment by talhahwahla** [comment]
Testing in v6 env

```bash
lookup result: IpDetails { ip: "2a01:4f9:c010:a3ff::1", hostname: None, city: "Helsinki", region: "Uusimaa", country: "FI", country_name: Some("Finland"), is_eu: Some(true), country_flag: Some(CountryFlag { emoji: "🇫🇮", unicode: "U+1F1EB U+1F1EE" }), country_flag_url: Some("https://cdn.ipinfo.io/static/images/countries-flags/FI.svg"), country_currency: Some(CountryCurrency { code: "EUR", symbol: "€" }), continent: Some(Continent { code: "EU", name: "Europe" }), loc: "60.1695,24.9354", org: Some("AS24940 Hetzner Online GmbH"), postal: Some("00100"), timezone: Some("Europe/Helsinki"), asn: Some(AsnDetails { asn: "AS24940", name: "Hetzner Online GmbH", domain: "hetzner.de", route: "2a01:4f9::/32", asn_type: "hosting" }), company: Some(CompanyDetails { name: "Hetzner Online GmbH", domain: "hetzner.com", company_type: "hosting" }), carrier: None, privacy: Some(PrivacyDetails { vpn: false, proxy: false, tor: false, relay: false, hosting: true, service: "" }), abuse: Some(AbuseDetails { address: "Hetzner Online GmbH, Industriestrasse 25, D-91710 Gunzenhausen, Germany", country: "DE", email: "abuse@hetzner.com", name: "Hetzner Online GmbH - Contact Role", network: "2a01:4f9:c010::/48", phone: "+49 9831 505-0" }), domains: Some(DomainsDetails { ip: None, total: 0, domains: [] }), bogon: None, extra: {} }

File: src/ipinfo.rs [hunk 1]
```diff
@@ -303,6 +303,19 @@ impl IpInfo {
     /// }
     /// ```
     pub async fn lookup(&mut self, ip: &str) -> Result<IpDetails, IpError> {
+        self._lookup(ip, BASE_URL).await
+    }
+
+    // Lookup your own v6 IP
```
Comment (Alextopher) [inline src/ipinfo.rs:309]: Can this be made into a doc comment? With sufficient explanation of purpose and limitations.

**Comment by talhahwahla** [comment]
I think I incorrectly concluded [here](https://github.com/ipinfo/rust/pull/52#issuecomment-1897960718) that we are allowing client level configurability in our Go SDK; doesn't seem to be the case after a closer look. So going with the suggestion [here](https://github.com/ipinfo/rust/pull/52#issuecomment-1896445160) to just remove input parameter and rename. @UmanShahzad

**Review by Alextopher — COMMENTED** [review]
If we're adding `lookup_self_v6` can we add the mirrored `lookup_self_v4`?

**Comment by talhahwahla** [comment]
cc @Alextopher

**Review by UmanShahzad — APPROVED** [review]

**Pull request merged**